### DISCUSSION TITLE
フルスキャンの方が早くなる理由

### DISCUSSION THREAD

**Discussion by yuyakawai0411** [kakeru-one/web-engineering-roundtable#20] [Q&A] [answered]
フルスキャンの場合は、一度にブロックを取ってくる。
フルスキャンじゃない場合は、ツリーで条件分岐したのちに、ブロックを取ってくる。
取ってくるブロック数が多いと、ブロックを個々に取得するため、オーバーヘッドが大きくなり、フルスキャンの方が早いことが起きうる

**Comment by kakeru-one** [comment, marked as answer]
## 根拠となる参考文献
Oracleでいう「ブロック」とは、MySQLで言う「ページ」のことである。
- https://doppo1.net/oracle/tuning/index_outline.html
- https://doppo1.net/oracle/tuning/index_outline.html#:~:text=%E5%90%91%E4%B8%8A%E3%81%97%E3%81%BE%E3%81%99%E3%80%82-,%E3%83%86%E3%83%BC%E3%83%96%E3%83%AB%E3%83%95%E3%83%AB%E3%82%B9%E3%82%AD%E3%83%A3%E3%83%B3%E3%81%AE%E3%81%BB%E3%81%86%E3%81%8C%E6%97%A9%E3%81%84%E5%A0%B4%E5%90%88,-%E4%B8%8A%E8%A8%98%E3%82%B1%E3%83%BC%E3%82%B9%E3%81%A7%E3%81%AF
